src/popsift/sift_conf.h

```diff
@@ -15,6 +15,13 @@
 
 #undef USE_DOG_TEX_LINEAR
 
+#ifdef _MSC_VER
+#define DEPRECATED(func) __declspec(deprecated) func
+#elif defined(__GNUC__) || defined(__clang__)
+#define DEPRECATED(func) func __attribute__ ((deprecated))
+#else
```

Comment:
We should have an empty definition as a last resort?
`#define DEPRECATED(func)`